pkg/guard/audit_logger.go

```diff
@@ -0,0 +1,56 @@
+package guard
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/justtrackio/gosoline/pkg/log"
+	"github.com/ory/ladon"
+)
+
+//go:generate mockery --name AuditLogger
+type AuditLogger interface {
+	ladon.AuditLogger
+}
+
+type auditLogger struct {
+	logger log.Logger
+}
+
+func NewAuditLogger(logger log.Logger) AuditLogger {
+	return &auditLogger{
+		logger: logger.WithChannel("guard_access").WithContext(context.Background()),
+	}
+}
+
+func (a auditLogger) LogRejectedAccessRequest(request *ladon.Request, pool ladon.Policies, deciders ladon.Policies) {
+	logger := a.logger.WithFields(buildLogFields(request))
+
+	if len(deciders) == 0 {
+		logger.Info("no policy allowed access")
+		return
+	}
+
+	rejecter := deciders[len(deciders)-1]
+
+	logger.Info("policy %s denied the access", rejecter.GetID())
+}
```

Comment:
what if I want to identify the policies that grant access? it appears that the final policy is the one that refuses the request. i would like to view the ids of the policies that permit access.